Cutelyst/Plugins/Utils/Sql/sql.cpp

```diff
@@ -94,6 +94,34 @@ QVariantList Sql::queryToMapList(QSqlQuery &query)
     return ret;
 }
 
+QVariantMap Sql::queryToMap(QSqlQuery &query, const QString &key)
+{
+    QVariantMap ret;
+    QSqlRecord record = query.record();
+
+    if (!record.contains(key)) {
+      qCCritical(C_SQL) << "Field Name " << key << " not found in result set";
+      return ret;
+    }
+
+    int columns = record.count();
+    QStringList cols;
+
+    for (int i = 0; i < columns; ++i) {
+      cols.append(record.fieldName(i));
+    }
+
+    while (query.next()) {
+      QVariantHash h;
```

Comment:
The problem with a Hash is that the order of insertMulti is random, in all of my tests QHash is faster than QMap, so we could have a queryToIndexedHash as well, as:
`{{ projects.main.title }}`
will work either way.